pkg/reconciler/v1alpha1/revision/reconcile_resources.go

```diff
@@ -108,10 +113,22 @@ func (c *Reconciler) reconcileKPA(ctx context.Context, rev *v1alpha1.Revision) e
 	} else if getKPAErr != nil {
 		logger.Errorf("Error reconciling kpa %q: %v", kpaName, getKPAErr)
 		return getKPAErr
+	} else {
+		// Preserve Generation
+		desiredPa.Spec.Generation = pa.Spec.Generation
+		if !equality.Semantic.DeepEqual(desiredPa.Spec, pa.Spec) || !equality.Semantic.DeepEqual(desiredPa.Annotations, pa.Annotations) {
+			logger.Info("Updating PA %q", desiredPa.Name)
+			pa.Spec = desiredPa.Spec
+			pa.Annotations = desiredPa.Annotations
+			if _, err := c.updateKPA(ctx, pa); err != nil {
```

Comment:
This will update the annotations.  We'd discussed making class annotations immutable on the PodAutoscaler (we should probably do this @ Revision as well), which simplifies this a bit.

Reply:
Correct.  I want the `class` and `metric` annotations to be immutable.  But I want the others to be reconciled (`minScale`, `maxScale`, `target`).

My plan was to just reconcile all annotations from Revision to PodAutoscaler (for simplicity).  And to use the Revision webhook to disallow `class` and `metric` changes.  That means I will also be adding defaults in the Revision webhook.

Reply:
After chatting with @mattmoor, I'll make the `class` annotation immutable on the Revision (enforced through webhook).  But if I can't allow `metric` to change, I'll give an error from the PodAutoscaler webhook or reconciler, so I don't leak implementation-specific annotations into higher layers.